app/controllers/application_controller.rb

```diff
@@ -1,35 +1,17 @@
-require 'json_envelope'
-
 class ApplicationController < ActionController::API
-  # when HTTP_REFERER exists, it's a great way to prevent CSRF attacks.
-  #
-  # an experiment performed in http://seclab.stanford.edu/websec/csrf/csrf.pdf found the
-  # header existed for 99.9% of users over HTTPS, even cross-origin. the header appears
-  # to be primarily suppressed at the network level, not the user agent.
-  #
-  # if this is ever determined insufficient, the backup plan is a custom header set by
-  # compatible javascript. stay stateless!
-  private def require_trusted_referrer
-    return if Rails.application.config.application_domains.include?(requesting_audience)
-    render status: :forbidden, json: JSONEnvelope.errors('referer' => 'is not a trusted host')
-  end
+  include AccessControl
 
-  private def require_api_credentials
-    auth_strategy = ActionController::HttpAuthentication::Basic
+  # Unauthorized access means credentials are required but absent.
+  class AccessUnauthorized < StandardError; end
```

Comment:
Hey, welcome!

I picked these names to mimic the 401 and 403 HTTP codes, but you're quite right that it's still potentially confusing. Will have to give it some thought and maybe consider avoiding those words altogether.